src/main/java/org/wildfly/security/auth/login/SecurityIdentity.java

```diff
@@ -169,4 +170,13 @@ public void runAs(Runnable action) {
             securityDomain.setCurrentSecurityIdentity(old);
         }
     }
+
+    /**
+     * <p>Get the roles associated with this identity.</p>
+     *
+     * @return the roles.
+     */
+    public Set<String> getRoles() {
```

Comment:
As discussed, we should have some realm info structure stored on the SI which would contain a reference to the realm and its mapper(s).